src/systemdhook.c

```diff
@@ -116,22 +116,26 @@ static int bind_mount(const char *src, const char *dest, int readonly) {
 	return 0;
 }
 
-static int chcon(const char *path, const char *label) {
+static int chperm(const char *path, const char *label, int uid, int gid) {
 	DIR *dir;
 	struct dirent *ent;
 	if ((dir = opendir (path)) != NULL) {
 		/* print all the files and directories within directory */
 		while ((ent = readdir (dir)) != NULL) {
 			_cleanup_free_ char *full_path = NULL;
 			if (asprintf(&full_path, "%s/%s", path, ent->d_name) < 0) {
-				pr_perror("Failed to create path for chcon");
+				pr_perror("Failed to create path for chperm");
 				closedir(dir);
 				return -1;
 
 			}
 			if (setfilecon (full_path, label) < 0) {
 				pr_perror("Failed to set context %s on %s", label, path);
 			}
+			/* Change uid and gid to something the container can handle */
+			if (chown(full_path, uid, gid) == -1) {
+				pr_perror("Failed to chown %d,%d to full_path owner: %s, errno: [%d]", uid, gid, full_path, errno);
+			}
```

Comment:
Dan you asked for a return -1 here, but if I do that, it will stop processing and won't create the /etc/machine-id.  It fails every time when the systemd-named-path is sent up here with a 30 error, read only file system.  I'm wondering if I should handle the 30 error as the pass through or just not do the chown on the systemd_full_path to begin with.  Still mulling/testing, thoughts gratefully accepted.